### DISCUSSION TITLE
Create a template in GitHub template repository

### DISCUSSION THREAD

**Discussion by Nelson-numerical-software** [mathworks/toolboxdesign#6] [Ideas]
A template repository could be easily reusuable by users
see https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository
Nice documentation, inspiring for me :p

**Comment by Nelson-numerical-software** [comment]
Something as toolbox skeleton/template reusable could be really great for users and communities
example: https://github.com/Nelson-numerical-software/module_skeleton (no pub here, feel free to remove)

**Comment by rpurser47** [comment]
Glad you like it!   Unfortunately, any template would have to generate unique UUIDs at creation time, so we can't just copy a template repo.   However, we are exploring a couple of ways of doing this, and right now, [cookiecutter](https://cookiecutter.readthedocs.io/en/stable/) is our current best candidate.  That way, people can create toolboxes from outside MATLAB, and we'd probably offer a simple interface from inside MATLAB.

**Reply by Nelson-numerical-software** [reply]
https://github.com/cargo-generate/cargo-generate ;)
